web-ui/src/components/reviews/TeamReviews.jsx

```diff
@@ -142,17 +157,33 @@ const TeamReviews = ({ onBack, periodId }) => {
   const reviewAssignmentsUrl = '/services/review-assignments';
 
   useEffect(() => {
-    loadTeamMembers();
+    loadAssignments();
   }, [currentMembers]);
 
+  useEffect(() => {
+    const myId = currentUser?.id;
+    const supervisors = selectSupervisors(state);
+    const isManager = supervisors.some(s => s.id === myId);
+    const period = selectReviewPeriod(state, periodId);
+    if (period) {
+      setApprovalMode(
+        isManager && period.reviewStatus === ReviewStatus.AWAITING_APPROVAL
+      );
+    }
+  }, [state]);
+
+  useEffect(() => {
+    loadTeamMembers();
+  }, [approvalMode, assignments, showAll]);
```

Comment:
We need to reload the team members to be displayed if any of these things change.